### PR TITLE
[ISSUE #5462] Remove RemotingCommand.decode parse header repeated calculation

### PR DESCRIPTION
**Pull request by liweiyuan**
**Make sure set the target branch to `develop`**

## What is the purpose of the change

XXXXX

## Brief changelog

- remove repeated calculation in RemotingCommand.decode

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java [hunk 1]
```diff
@@ -181,13 +181,12 @@ public static RemotingCommand decode(final ByteBuffer byteBuffer) throws Remotin
 
     public static RemotingCommand decode(final ByteBuf byteBuffer) throws RemotingCommandException {
         int length = byteBuffer.readableBytes();
-        int oriHeaderLen = byteBuffer.readInt();
-        int headerLength = getHeaderLength(oriHeaderLen);
+        int headerLength = byteBuffer.readInt();
```
Comment (lizhanhui) [inline remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java:184]: According to the frame layout, the first byte of the header length is used as a bit-field, flagging how the header is serialized.  Only the following 3 bytes are actually employed to represent header length.

**Review by lizhanhui — CHANGES_REQUESTED** [review]
Logically inconsistent with existing design

File: remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java (hunk 1 above)
Comment (liweiyuan) [reply remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java:185]: to optimize the code， support get protocol type from ByteBuf

**Comment by lizhanhui** [comment]
Rebase to make CI pass first.

**Comment by liweiyuan** [comment]
ok, have to rebase code .

**Comment by liweiyuan** [comment]
Again rebase code

**Comment by tsunghanjacktsai** [comment]
Hey @liweiyuan , seems like you've sent multiple duplicated commits, be sure to squash them to keep your commit logs clean.

**Review by tsunghanjacktsai — CHANGES_REQUESTED** [review]
Hey @liweiyuan , seems like you've sent multiple duplicated commits, be sure to squash them to keep your commit logs clean.

**Comment by liweiyuan** [comment]
ok,@tsunghanjacktsai

**Pull request closed without merging**